### PR TITLE
Brandon mvn repo

### PR DESCRIPTION
**Pull request by branflake2267**
I got the the maven repo to build. but seems I committed some classpath files in this. But I thought I'd pull it for review. This would make it so we could distribute the maven repo through this git hub repo.

https://github.com/branflake2267/GWT-Maps-V3-Api/tree/brandon_mvn_repo/distribution/snapshots/com/google/gwt/maps/Apis_Google_Maps

### CODE REVIEW

File: Apis_Maps_Test/.project [hunk 1]
```diff
@@ -1,40 +1,53 @@
...
+		<nature>org.eclipse.jdt.core.javanature</nature>
+		<nature>org.eclipse.wst.common.project.facet.core.nature</nature>
+		<nature>com.google.appengine.eclipse.core.gaeNature</nature>
+		<nature>com.google.gwt.eclipse.core.gwtNature</nature>
+	</natures>
+	<linkedResources>
+		<link>
+			<name>src_maps</name>
+			<type>2</type>
+			<location>/Users/branflake2267/git/GWT-Maps-V3-Api/Apis_Google_Maps/src</location>
```
Comment (branflake2267) [inline Apis_Maps_Test/.project:50]: oops

File: distribution/snapshots/com/google/gwt/maps/Apis_Google_Maps/3.8.0-beta2-SNAPSHOT/Apis_Google_Maps-3.8.0-beta2-20120916.232453-1.pom [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+    <!-- To deploy into the my maven repo use command on next line ran in the pom.xml directory -->
+    <!-- ./"mvn -DaltDeploymentRepository=snapshots::default::file:../distribution/snapshots/ clean deploy" -->
+    <distributionManagement>
+    <repository>
+      <id>releases</id>
+      <url>file:../distribution/releases/</url>
+    </repository>
+      <snapshotRepository>
+        <id>snapshots</id>
+        <url>file:../distribution/snapshots/</url>
```
Comment (branflake2267) [inline distribution/snapshots/com/google/gwt/maps/Apis_Google_Maps/3.8.0-beta2-SNAPSHOT/Apis_Google_Maps-3.8.0-beta2-20120916.232453-1.pom:38]: This deploys to git repo

File: distribution/snapshots/com/google/gwt/maps/Apis_Google_Maps/3.8.0-beta2-SNAPSHOT/Apis_Google_Maps-3.8.0-beta2-20120916.232453-1.pom [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+		<artifactId>gwt-maps-api-v3</artifactId>
+		<version>3.8.0-beta2-SNAPSHOT</version>
+	</parent>
+
+	<properties>
+		<webappDirectory>${project.build.directory}/${project.build.finalName}</webappDirectory>
+	</properties>
+  
+    <!-- To deploy into the my maven repo use command on next line ran in the pom.xml directory -->
+    <!-- ./"mvn -DaltDeploymentRepository=snapshots::default::file:../distribution/snapshots/ clean deploy" -->
```
Comment (branflake2267) [inline distribution/snapshots/com/google/gwt/maps/Apis_Google_Maps/3.8.0-beta2-SNAPSHOT/Apis_Google_Maps-3.8.0-beta2-20120916.232453-1.pom:30]: I don't have to use this now. instead its mvn clean deploy

**Comment by branflake2267** [comment]
@twistedpair Would this type of deployment throw sand in the gears?

**Comment by twistedpair** [comment]
I've seen other projects use this poor man's repo before. It should work for now, and since it is not a personal server hosting it, but rather GitHub, the overhead should not be a concern. However, the greater implementation is to have the artifacts in Maven Central so that people don't have to add the GitHub repo. This is a good stopgap in the mean time to ease update. I assume we would have a 3.8.x-SNAPHOT and do 3.8.++ for notable bug fixes?

**Comment by branflake2267** [comment]
Nicely said. :) I'm up for whatever it works. Whats the process for getting it into the maven central, is that time consuming?

**Comment by branflake2267** [comment]
https://docs.sonatype.org/display/Repository/Sonatype+OSS+Maven+Repository+Usage+Guide#SonatypeOSSMavenRepositoryUsageGuide-2.Signup - just looking at how to sign up.

**Comment by branflake2267** [comment]
I'm adding a new ticket to ask for project hosting.

**Comment by branflake2267** [comment]
https://issues.sonatype.org/browse/OSSRH-4661 - filed issue

**Comment by twistedpair** [comment]
Nice work.

**Comment by branflake2267** [comment]
Thanks :)

**Pull request closed without merging**